ttcn3/types/types.go

```diff
@@ -179,34 +179,42 @@ type ListType struct {
 }
 
 func (t *ListType) String() string {
+	elem := "any"
+	if t.ElementType != nil && !strings.Contains(t.Kind.String(), "string") {
```

Comment:
Comparing with a string is effective, but fragile. I suggest you check recursive types explicitly using the Kind or a helper function